src/inmanta/parser/plyInmantaLex.py

```diff
@@ -132,7 +134,24 @@ def t_JCOMMENT(t: lex.LexToken) -> None:  # noqa: N802
 
 def t_MLS(t: lex.LexToken) -> lex.LexToken:
     r'"{3,5}([\s\S]*?)"{3,5}'
-    value = bytes(t.value[3:-3], "utf-8").decode("unicode_escape")
+
+    #   Check for the presence of an invalid escape sequence (ex: "\.")
+    #   Python < 3.12 raises a DeprecationWarning when encountering an invalid escape sequence
+    #   Python 3.12 will raise a SyntaxWarning
+    #   Future versions will eventually raise a SyntaxError
+    #   see ( https://docs.python.org/3.12/whatsnew/3.12.html#other-language-changes )
+    try:
+        with warnings.catch_warnings():
+            warnings.filterwarnings("error", message="invalid escape sequence")
+            value: str = bytes(t.value[3:-3], "utf_8").decode("unicode_escape")
+    except DeprecationWarning:
+        with warnings.catch_warnings():
+            warnings.filterwarnings("ignore")
+            value = bytes(typing.cast(str, t.value)[3:-3], "utf_8").decode("unicode_escape")
+
+        msg: str = f"({t.lexer.inmfile}:{t.lexer.lineno}) Invalid escape sequence in multi-line string:\n{value}"
+        warnings.showwarning(msg, lineno=t.lexer.lineno, filename=t.lexer.inmfile, category=ParserWarning)
```

Comment:
Why use the low-level `showwarning` rather than `warnings.warn(ParserWarning(...))`?

Reply:
I had 2 issues:
- I couldn't find a clean way to get the file name and line number in there using `Range()` but looking at it again, I think I can use `Location()`
- As it is a subclass of an `InmantaWarning`, the value itself wasn't properly displayed, but I agree with your other comment that it's preferable not to show the string in the warning message.

Conclusion -> I'll use `warnings.warn(ParserWarning(...))`